worker/src/synchronizer.rs

```diff
@@ -164,7 +162,7 @@ impl Synchronizer {
 
                             // Add the digest to the waiter.
                             let deliver = digest;
-                            let (tx_cancel, rx_cancel) = channel(1);
+                            let (tx_cancel, rx_cancel) = mpsc::channel(1);
```

Comment:
Ah probably those could be oneshot channels? (not concern of this PR of course - but something to keep in mind)

Reply:
They usually can't: this is a cancellation handler, and the branches of the `tokio::select` that GC and those that unpack items in waiting race each other. The GC might then try to cancel an item that's already delivered (and the delivery sinks the receiving end of the channel). With a oneshot channel, that may lead to a panic.